### PR TITLE
Move to pref branch for ASR

### PR DESCRIPTION
**Pull request by k88hudson**
This patch moves our various configurations to a pref branch for ASR rather than a single prefs to support the shield roll-out mechanism. It also turns on snippets in nightly and CFR for all channels.

Please test with the following steps:

1. With a temp-profile/new profile, turn on the devtoolsEnabled pref and visit about:newtab#asrouter
2. Ensure CFR is on by default and the 3-lifetime cap variant is the default
3. Try turning different providers on/off.
4. With ASR snippets disabled, ensure legacy snippets are properly loaded in about:newtab by opening the console.
5. With ASR snippets enabled, ensure you see actual content on new tab pages.

### CODE REVIEW

File: content-src/asrouter/README.md [hunk 1]
```diff
@@ -5,24 +5,29 @@
 Name | Used for | Type | Example value
 ---  | ---      | ---  | ---
 `whitelistHosts` | Whitelist a host in order to fetch messages from its endpoint | `[String]` |  `["gist.github.com", "gist.githubusercontent.com", "localhost:8000"]`
-`messageProviders` | Message provider options | `Object` | [see below](#message-providers)
+`providers.snippets` | Message provider options for snippets | `Object` | [see below](#message-providers)
+`providers.cfr` | Message provider options for cfr | `Object` | [see below](#message-providers)
+`providers.onboarding` | Message provider options for cfr | `Object` | [see below](#message-providers)
```
Comment (sarracini) [inline content-src/asrouter/README.md]: copy pasta: `cfr` should be `onboarding`

File: lib/ActivityStream.jsm [hunk 2]
```diff
@@ -202,33 +202,34 @@ const PREFS_CONFIG = new Map([
...
+    title: "Does the user allow CFR recommendations?",
+    value: JSON.stringify({
       id: "cfr",
       type: "local",
       localProvider: "CFRMessageProvider",
-      enabled: IS_NIGHTLY_OR_UNBRANDED_BUILD,
-      cohort: IS_NIGHTLY_OR_UNBRANDED_BUILD ? "nightly" : "",
-    }]),
+      frequency: {custom: [{period: "daily", cap: 1}]},
+      enabled: true,
```
Comment (sarracini) [inline lib/ActivityStream.jsm]: Why is the frequency here now?

File: lib/ActivityStream.jsm [hunk 3]
```diff
@@ -202,33 +202,34 @@ const PREFS_CONFIG = new Map([
...
-
-    /**
-     * Each provider must have a unique id and a type of "local" or "remote".
-     * Local providers must specify the name of an ASRouter message provider.
-     * Remote providers must specify a `url` and an `updateCycleInMs`.
-     * Each provider must also have an `enabled` boolean.
-     */
-    value: JSON.stringify([{
+  ["asrouter.providers.onboarding", {
+    title: "Does the user allow CFR recommendations?",
```
Comment (sarracini) [inline lib/ActivityStream.jsm]: Title should be for onboarding

File: lib/ActivityStream.jsm [hunk 4]
```diff
@@ -202,33 +202,34 @@ const PREFS_CONFIG = new Map([
...
+    value: JSON.stringify({
       id: "onboarding",
       type: "local",
       localProvider: "OnboardingMessageProvider",
       enabled: true,
-    }, {
+    }),
+  }],
+  ["asrouter.providers.snippets", {
+    title: "Does the user allow CFR recommendations?",
```
Comment (sarracini) [inline lib/ActivityStream.jsm]: Title should be for snippets

File: lib/ActivityStream.jsm [hunk 5]
```diff
@@ -202,33 +202,34 @@ const PREFS_CONFIG = new Map([
...
       type: "remote",
       url: "https://snippets.cdn.mozilla.net/%STARTPAGE_VERSION%/%NAME%/%VERSION%/%APPBUILDID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/",
       updateCycleInMs: ONE_HOUR_IN_MS * 4,
-      enabled: false,
-    }, {
+      enabled: UpdateUtils.getUpdateChannel(true) === "nightly",
+    }),
+  }],
+  ["asrouter.providers.cfr", {
+    title: "Does the user allow CFR recommendations?",
```
Comment (sarracini) [inline lib/ActivityStream.jsm]: This is the title for `asrouter.userprefs.cfr`

File: lib/CFRMessageProvider.jsm [hunk 6]
```diff
@@ -345,7 +343,6 @@ const CFR_MESSAGES = [
     frequency: {lifetime: 3},
     targeting: `
       localeLanguageCode == "en" &&
-      (providerCohorts.cfr == "three_per_day") &&
```
Comment (sarracini) [inline lib/CFRMessageProvider.jsm]: I'm a bit confused with this cohort business.. how come some of them still have cohorts but the config pref doesn't have cohort anymore?

File: lib/ActivityStream.jsm (hunk 2 above)
Comment (k88hudson) [reply lib/ActivityStream.jsm]: Before it was in the experiment definition here: https://gist.github.com/k88hudson/8e1fb4a7a84cd521832aaf36cb6f62e3 now that it's getting shipped to production, it needs to be in the message definition.

File: lib/CFRMessageProvider.jsm (hunk 6 above)
Comment (k88hudson) [reply lib/CFRMessageProvider.jsm]: I was just going to leave the cohort in case we wanted to test more with it, I guess I could just remove it though

**Comment by rlr** [comment]
I *think* this is patch causing a test failure: `GECKO(7922) | FATAL ERROR: Non-local network connections are disabled and a connection attempt to snippets.cdn.mozilla.net (52.84.230.7) was made.`

https://treeherder.mozilla.org/#/jobs?repo=mozilla-central&resultStatus=success%2Ctestfailed%2Cbusted%2Cexception&classifiedState=unclassified&searchStr=nightly%2Cmochitest&selectedJob=210989036&revision=e9ff3fec51d3de454f607b69b4acddc2a81409c1&group_state=expanded

**Pull request merged**